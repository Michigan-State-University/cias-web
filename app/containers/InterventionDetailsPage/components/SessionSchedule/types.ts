import { SessionSchedule } from 'models/Session';

export type SessionScheduleOption = { id: SessionSchedule; label: string };
