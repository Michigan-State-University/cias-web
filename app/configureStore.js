/**
 * Create the store with dynamic reducers
 */

import { createStore, applyMiddleware, compose } from 'redux';
import { routerMiddleware } from 'connected-react-router';
import createSagaMiddleware from 'redux-saga';

import utilsHistory from 'utils/history';
import { initialState as authReducerInitialState } from 'global/reducers/auth/reducer';

import { loadState } from 'utils/persist';
import { createInjectorsEnhancer } from 'redux-injectors';
import createReducer from './reducers';

export default function configureStore(initialState = {}, history) {
  let composeEnhancers = compose;
  const reduxSagaMonitorOptions = {};

  // If Redux Dev Tools and Saga Dev Tools Extensions are installed, enable them
  /* istanbul ignore next */
  if (process.env.NODE_ENV !== 'production' && typeof window === 'object') {
    /* eslint-disable no-underscore-dangle */
    if (window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__)
      composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
        serialize: {
          options: {
            map: true,
          },
        },
      });

    // NOTE: Uncomment the code below to restore support for Redux Saga
    // Dev Tools once it supports redux-saga version 1.x.x
    // if (window.__SAGA_MONITOR_EXTENSION__)
    //   reduxSagaMonitorOptions = {
    //     sagaMonitor: window.__SAGA_MONITOR_EXTENSION__,
    //   };
    /* eslint-enable */
  }

  const sagaMiddleware = createSagaMiddleware(reduxSagaMonitorOptions);

  // Create the store with two middlewares
  // 1. sagaMiddleware: Makes redux-sagas work
  // 2. routerMiddleware: Syncs the location/URL path to the state
  const middlewares = [sagaMiddleware, routerMiddleware(history)];

  const runSaga = sagaMiddleware.run;

  const enhancers = [
    applyMiddleware(...middlewares),
    createInjectorsEnhancer({
      createReducer,
      runSaga,
    }),
  ];

  const store = createStore(
    createReducer(),
    initialState,
    composeEnhancers(...enhancers),
  );

  // Extensions
  store.injectedReducers = {}; // Reducer registry
  store.injectedSagas = {}; // Saga registry

  // Make reducers hot reloadable, see http://mxs.is/googmo
  /* istanbul ignore next */
  if (module.hot) {
    module.hot.accept('./reducers', () => {
      store.replaceReducer(store.createReducer(store.injectedReducers));
    });
  }

  return store;
}
const { user } = loadState() ?? {};

const initialStore = { auth: { ...authReducerInitialState, user } };

export const store = configureStore(initialStore, utilsHistory);
