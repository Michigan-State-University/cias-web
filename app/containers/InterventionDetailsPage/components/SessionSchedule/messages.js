/*
 * SessionSchedule Messages
 *
 * This contains all the text for the SessionSchedule component.
 */

import { defineMessages } from 'react-intl';

import { SessionSchedule } from 'models/Session';

export const scope = 'app.components.SessionSchedule';

export default defineMessages({
  info: {
    id: `${scope}.info`,
    defaultMessage: 'If this session is next in the sequence',
  },
  sessionSchedulingHelp: {
    id: `${scope}.sessionSchedulingHelp`,
    defaultMessage: `Schedule when each session of your intervention is sent to participants. Learn more <a href='https://www.cias.app/_files/ugd/afc5c9_44a2c4b39b5e431daf98acdb4d61aa36.pdf' target='_blank'>here</a>. For more help resources, visit <a href='https://www.cias.app/resources' target='_blank'>www.cias.app/resources</a>.`,
  },
  send: {
    id: `${scope}.send`,
    defaultMessage: 'Send ',
  },
  [SessionSchedule.DAYS_AFTER]: {
    id: `${scope}.${SessionSchedule.DAYS_AFTER}`,
    defaultMessage:
      'Send email invitation [X] days after previous session scheduled date',
  },
  [SessionSchedule.DAYS_AFTER_FILL]: {
    id: `${scope}.${[SessionSchedule.DAYS_AFTER_FILL]}`,
    defaultMessage:
      'Send email invitation [X] days after previous session completed date',
  },
  [SessionSchedule.DAYS_AFTER_DATE]: {
    id: `${scope}.${[SessionSchedule.DAYS_AFTER_DATE]}`,
    defaultMessage:
      'Send email invitation [X] days after a date provided by participant',
  },
  [SessionSchedule.EXACT_DATE]: {
    id: `${scope}.${[SessionSchedule.EXACT_DATE]}`,
    defaultMessage: 'Send email invitation on exact date',
  },
  [SessionSchedule.AFTER_FILL]: {
    id: `${scope}.${[SessionSchedule.AFTER_FILL]}`,
    defaultMessage:
      'Send email invitation immediately after previous session completion',
  },
  [SessionSchedule.IMMEDIATELY]: {
    id: `${scope}.${SessionSchedule.IMMEDIATELY}`,
    defaultMessage:
      'Start session immediately after previous session completion',
  },
  daysAfterInfo: {
    id: `${scope}.daysAfterInfo`,
    defaultMessage: 'days after previous session scheduled date.',
  },
  daysAfterFillInfo: {
    id: `${scope}.daysAfterFillInfo`,
    defaultMessage: 'days after previous session completed date.',
  },
  daysAfterDateInfo: {
    id: `${scope}.daysAfterDateInfo`,
    defaultMessage: 'days after a date provided by participant',
  },
  exactDateInfo: {
    id: `${scope}.exactDateInfo`,
    defaultMessage: 'on exact date',
  },
  enterNumber: {
    id: `${scope}.enterNumber`,
    defaultMessage: 'Enter number',
  },
  default: {
    id: `${scope}.default`,
    defaultMessage: 'Select option',
  },
  timeInfo: {
    id: `${scope}.timeInfo`,
    defaultMessage: '*The information will be sent at',
  },
  daysAfterDateVariableEmpty: {
    id: `${scope}.daysAfterDateVariableEmpty`,
    defaultMessage: 'Choose variable',
  },
  daysAfterDateVariableInfo: {
    id: `${scope}.daysAfterDateVariableInfo`,
    defaultMessage: '(Date Screen variable which should be used)',
  },
});
