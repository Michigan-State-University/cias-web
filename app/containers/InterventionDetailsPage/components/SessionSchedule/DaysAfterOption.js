import React, { useMemo } from 'react';
import PropTypes from 'prop-types';
import { injectIntl } from 'react-intl';

import { numericValidator } from 'utils/validators';
import isNullOrUndefined from 'utils/isNullOrUndefined';
import { SCHEDULE_OPTIONS } from 'global/reducers/intervention';

import ApprovableInput from 'components/Input/ApprovableInput';
import Text from 'components/Text';

import messages from './messages';
import { StyledInputWrapper } from './styled';

const DaysAfterOption = ({
  intl: { formatMessage },
  value,
  setValue,
  disabled,
  scheduleOption,
}) => {
  const getValue = () => (isNullOrUndefined(value) ? '' : value);

  const textInfo = useMemo(() => {
    switch (scheduleOption) {
      case SCHEDULE_OPTIONS.daysAfterFill:
        return formatMessage(messages.daysAfterFillInfo);
      case SCHEDULE_OPTIONS.daysAfterDate:
        return formatMessage(messages.daysAfterDateInfo);
      case SCHEDULE_OPTIONS.daysAfter:
      default:
        return formatMessage(messages.daysAfterInfo);
    }
  }, [scheduleOption]);

  return (
    <>
      <Text fontSize={15} wordBreak="normal">
        {formatMessage(messages.send)}
      </Text>
      <StyledInputWrapper>
        <ApprovableInput
          disabled={disabled}
          type="singleline"
          placeholder={formatMessage(messages.enterNumber)}
          validator={numericValidator}
          height={50}
          mr={0}
          value={getValue()}
          onCheck={number => setValue(number)}
          fontSize={15}
          padding={5}
          textAlign="center"
        />
      </StyledInputWrapper>
      <Text fontSize={15}>{textInfo}</Text>
    </>
  );
};

DaysAfterOption.propTypes = {
  intl: PropTypes.object,
  scheduleOption: PropTypes.string,
  value: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  setValue: PropTypes.func,
  disabled: PropTypes.bool,
};

export default injectIntl(DaysAfterOption);
