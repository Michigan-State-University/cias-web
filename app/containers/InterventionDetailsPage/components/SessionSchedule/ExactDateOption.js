import React from 'react';
import PropTypes from 'prop-types';
import { injectIntl } from 'react-intl';

import ApprovableInput from 'components/Input/ApprovableInput';
import Text from 'components/Text';
import Box from 'components/Box';

import { themeColors } from 'theme';
import calculateTimeZone from 'utils/calculateTimeZone';
import { getUTCTime, getUTCDateString } from 'utils/dateUtils';

import Row from 'components/Row';
import Column from 'components/Column';
import messages from './messages';

const ExactDateOption = ({
  intl: { formatMessage },
  value,
  setValue,
  disabled,
}) => {
  const timeInfoMessage = () =>
    `${formatMessage(messages.timeInfo)} ${calculateTimeZone()}`;

  const onChange = (changeValue) => setValue(getUTCDateString(changeValue));

  return (
    <Column>
      <Row align="center" gap={10}>
        <Text ml={5} fontSize={15}>
          {formatMessage(messages.send)}
          {formatMessage(messages.exactDateInfo)}
        </Text>
        <Box>
          <ApprovableInput
            disabled={disabled}
            width={130}
            height={50}
            type="date"
            value={value ? getUTCTime(new Date(value)) : null}
            onCheck={onChange}
            fontSize={15}
            minDate={new Date()}
          />
        </Box>
      </Row>
      <Row mt={10} width="100%" justify="end">
        <Text ml={5} fontSize={12} color={themeColors.warning}>
          {timeInfoMessage()}
        </Text>
      </Row>
    </Column>
  );
};

ExactDateOption.propTypes = {
  intl: PropTypes.object,
  value: PropTypes.oneOfType([PropTypes.instanceOf(Date), PropTypes.string]),
  setValue: PropTypes.func,
  disabled: PropTypes.bool,
};

export default injectIntl(ExactDateOption);
