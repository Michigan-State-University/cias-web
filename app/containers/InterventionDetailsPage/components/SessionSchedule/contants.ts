import { SessionSchedule } from 'models/Session';

import { formatMessage } from 'utils/intlOutsideReact';

import { SessionScheduleOption } from './types';
import messages from './messages';

export const SESSION_SCHEDULE_OPTIONS: Record<
  SessionSchedule,
  SessionScheduleOption
> = {
  [SessionSchedule.AFTER_FILL]: {
    id: SessionSchedule.AFTER_FILL,
    label: formatMessage(messages[SessionSchedule.AFTER_FILL]),
  },
  [SessionSchedule.DAYS_AFTER]: {
    id: SessionSchedule.DAYS_AFTER,
    label: formatMessage(messages[SessionSchedule.DAYS_AFTER]),
  },
  [SessionSchedule.DAYS_AFTER_FILL]: {
    id: SessionSchedule.DAYS_AFTER_FILL,
    label: formatMessage(messages[SessionSchedule.DAYS_AFTER_FILL]),
  },
  [SessionSchedule.DAYS_AFTER_DATE]: {
    id: SessionSchedule.DAYS_AFTER_DATE,
    label: formatMessage(messages[SessionSchedule.DAYS_AFTER_DATE]),
  },
  [SessionSchedule.EXACT_DATE]: {
    id: SessionSchedule.EXACT_DATE,
    label: formatMessage(messages[SessionSchedule.EXACT_DATE]),
  },
  [SessionSchedule.IMMEDIATELY]: {
    id: SessionSchedule.IMMEDIATELY,
    label: formatMessage(messages[SessionSchedule.IMMEDIATELY]),
  },
};
