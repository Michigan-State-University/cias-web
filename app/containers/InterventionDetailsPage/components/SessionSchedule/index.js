/**
 *
 * SessionSchedule
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { injectIntl } from 'react-intl';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { createStructuredSelector } from 'reselect';
import values from 'lodash/values';
import find from 'lodash/find';

import { dateQuestion } from 'models/Session/QuestionTypes';
import { SessionTypes } from 'models/Session';

import {
  SCHEDULE_OPTIONS,
  changeSchedulingType,
  updateSchedulingDate,
  updateSchedulingPayload,
  changeCurrentSession,
  updateDaysAfterDateVariable,
} from 'global/reducers/intervention';
import {
  getQuestionGroupsRequest,
  makeSelectQuestionGroupsSessionId,
} from 'global/reducers/questionGroups';

import { colors, themeColors } from 'theme';

import VariableChooser from 'containers/VariableChooser';

import Text from 'components/Text';
import Column from 'components/Column';
import Selector from 'components/Selector';
import Row from 'components/Row';
import Badge from 'components/Badge';

import ExactDateOption from './ExactDateOption';
import DaysAfterOption from './DaysAfterOption';
import messages from './messages';

function SessionSchedule({
  intl: { formatMessage },
  selectedScheduleOption,
  scheduleAt,
  changeType,
  updatePayload,
  updateDate,
  schedulePayload,
  daysAfterDateVariableName,
  sessionId,
  disabled,
  session,
  updateDateVariable,
}) {
  const { interventionId } = session ?? {};

  const scheduleOptions = {
    afterFill: {
      id: SCHEDULE_OPTIONS.afterFill,
      label: formatMessage(messages.afterFill),
    },
    daysAfter: {
      id: SCHEDULE_OPTIONS.daysAfter,
      label: formatMessage(messages.daysAfter),
    },
    daysAfterFill: {
      id: SCHEDULE_OPTIONS.daysAfterFill,
      label: formatMessage(messages.daysAfterFill),
    },
    daysAfterDate: {
      id: SCHEDULE_OPTIONS.daysAfterDate,
      label: formatMessage(messages.daysAfterDate),
    },
    exactDate: {
      id: SCHEDULE_OPTIONS.exactDate,
      label: formatMessage(messages.exactDate),
    },
  };

  const handleOnClickDateVariable = (value) =>
    updateDateVariable(value, sessionId);

  const handleChangeDate = (date) => updateDate(date, sessionId);
  const handleChangeDays = (days) => updatePayload(days, sessionId);

  const renderOption = () => {
    switch (selectedScheduleOption) {
      case scheduleOptions.daysAfter.id:
        return (
          <DaysAfterOption
            id={sessionId}
            value={schedulePayload}
            setValue={handleChangeDays}
            disabled={disabled}
            scheduleOption={selectedScheduleOption}
          />
        );
      case scheduleOptions.daysAfterFill.id:
        return (
          <DaysAfterOption
            id={sessionId}
            value={schedulePayload}
            setValue={handleChangeDays}
            disabled={disabled}
            scheduleOption={selectedScheduleOption}
          />
        );
      case scheduleOptions.daysAfterDate.id:
        return (
          <Column>
            <Row align="center">
              <DaysAfterOption
                id={sessionId}
                value={schedulePayload}
                setValue={handleChangeDays}
                disabled={disabled}
                scheduleOption={selectedScheduleOption}
              />
            </Row>
            <Row mt={10} align="center">
              <VariableChooser
                disabled={disabled}
                currentInterventionId={interventionId}
                onClick={handleOnClickDateVariable}
                placement="left"
                questionTypeWhitelist={[dateQuestion.id]}
                currentSessionId={sessionId}
                includeAllVariables
                includeCurrentSession={false}
                includeNonDigitVariables
                isMultiSession
                sessionTypesWhiteList={[SessionTypes.CLASSIC_SESSION]}
              >
                <Badge bg={themeColors.primary} color={colors.white}>
                  {daysAfterDateVariableName ??
                    formatMessage(messages.daysAfterDateVariableEmpty)}
                </Badge>
              </VariableChooser>
              <Text ml={5}>
                {formatMessage(messages.daysAfterDateVariableInfo)}
              </Text>
            </Row>
          </Column>
        );
      case scheduleOptions.exactDate.id:
        return (
          <ExactDateOption
            disabled={disabled}
            value={scheduleAt}
            setValue={handleChangeDate}
          />
        );
      case scheduleOptions.afterFill.id:
        return <></>;
      default:
        break;
    }
  };
  return (
    <Column>
      <Text mb={12} textOpacity={0.6} color={colors.bluewood} fontSize={13}>
        {formatMessage(messages.info)}
      </Text>
      <Selector
        disabled={disabled}
        selectOptionPlaceholder={formatMessage(messages.default)}
        options={values(scheduleOptions)}
        activeOption={find(
          scheduleOptions,
          (elem) => elem.id === selectedScheduleOption,
        )}
        rightPosition="315"
        setOption={(id) => changeType(id, sessionId)}
      />
      {selectedScheduleOption && (
        <Row mt={28} mb={17} align="center">
          {renderOption()}
        </Row>
      )}
    </Column>
  );
}

SessionSchedule.propTypes = {
  intl: PropTypes.object,
  selectedScheduleOption: PropTypes.string,
  scheduleAt: PropTypes.oneOfType([
    PropTypes.instanceOf(Date),
    PropTypes.string,
  ]),
  schedulePayload: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  daysAfterDateVariableName: PropTypes.string,
  changeType: PropTypes.func,
  updatePayload: PropTypes.func,
  updateDate: PropTypes.func,
  sessionId: PropTypes.string,
  disabled: PropTypes.bool,
  updateDateVariable: PropTypes.func,
  session: PropTypes.object,
};

const mapStateToProps = createStructuredSelector({
  activeSessionId: makeSelectQuestionGroupsSessionId(),
});

const mapDispatchToProps = {
  changeType: changeSchedulingType,
  updatePayload: updateSchedulingPayload,
  updateDate: updateSchedulingDate,
  updateDateVariable: updateDaysAfterDateVariable,
  changeSessionIndex: changeCurrentSession,
  fetchQuestions: getQuestionGroupsRequest,
};

const withConnect = connect(mapStateToProps, mapDispatchToProps);

export default compose(injectIntl, withConnect)(SessionSchedule);
