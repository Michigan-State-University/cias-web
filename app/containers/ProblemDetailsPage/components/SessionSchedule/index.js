/**
 *
 * SessionSchedule
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { injectIntl } from 'react-intl';
import { connect } from 'react-redux';
import { compose } from 'redux';
import values from 'lodash/values';
import find from 'lodash/find';

import Text from 'components/Text';
import Column from 'components/Column';
import Selector from 'components/Selector';
import Row from 'components/Row';

import { colors } from 'theme';

import {
  SCHEDULE_OPTIONS,
  changeSchedulingType,
  updateSchedulingDate,
  updateSchedulingPayload,
} from 'global/reducers/intervention';

import ExactDateOption from './ExactDateOption';
import DaysAfterOption from './DaysAfterOption';
import messages from './messages';

function SessionSchedule({
  intl: { formatMessage },
  selectedScheduleOption,
  scheduleAt,
  changeType,
  updatePayload,
  updateDate,
  schedulePayload,
  sessionId,
  disabled,
}) {
  const scheduleOptions = {
    daysAfter: {
      id: SCHEDULE_OPTIONS.daysAfter,
      label: formatMessage(messages.daysAfter),
    },
    daysAfterFill: {
      id: SCHEDULE_OPTIONS.daysAfterFill,
      label: formatMessage(messages.daysAfterFill),
    },
    exactDate: {
      id: SCHEDULE_OPTIONS.exactDate,
      label: formatMessage(messages.exactDate),
    },
  };

  const handleChangeDate = date => updateDate(date, sessionId);
  const handleChangeDays = days => updatePayload(days, sessionId);

  const renderOption = () => {
    switch (selectedScheduleOption) {
      case scheduleOptions.daysAfter.id:
        return (
          <DaysAfterOption
            id={sessionId}
            value={schedulePayload}
            setValue={handleChangeDays}
            disabled={disabled}
          />
        );
      case scheduleOptions.daysAfterFill.id:
        return (
          <DaysAfterOption
            id={sessionId}
            value={schedulePayload}
            setValue={handleChangeDays}
            disabled={disabled}
            afterFill
          />
        );
      case scheduleOptions.exactDate.id:
        return (
          <ExactDateOption
            disabled={disabled}
            value={scheduleAt}
            setValue={handleChangeDate}
          />
        );
      default:
        break;
    }
  };
  return (
    <Column>
      <Text mb={12} textOpacity={0.6} color={colors.bluewood} fontSize={13}>
        {formatMessage(messages.info)}
      </Text>
      <Selector
        disabled={disabled}
        selectOptionPlaceholder={formatMessage(messages.default)}
        options={values(scheduleOptions)}
        activeOption={find(
          scheduleOptions,
          elem => elem.id === selectedScheduleOption,
        )}
        rightPosition="315"
        setOption={id => changeType(id, sessionId)}
      />
      {selectedScheduleOption && (
        <Row mt={28} mb={17} align="center">
          {renderOption()}
        </Row>
      )}
    </Column>
  );
}

SessionSchedule.propTypes = {
  intl: PropTypes.object,
  selectedScheduleOption: PropTypes.string,
  scheduleAt: PropTypes.oneOfType([
    PropTypes.instanceOf(Date),
    PropTypes.string,
  ]),
  schedulePayload: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  changeType: PropTypes.func,
  updatePayload: PropTypes.func,
  updateDate: PropTypes.func,
  sessionId: PropTypes.string,
  disabled: PropTypes.bool,
};

const mapDispatchToProps = {
  changeType: changeSchedulingType,
  updatePayload: updateSchedulingPayload,
  updateDate: updateSchedulingDate,
};

const withConnect = connect(
  null,
  mapDispatchToProps,
);

export default compose(
  injectIntl,
  withConnect,
)(SessionSchedule);
