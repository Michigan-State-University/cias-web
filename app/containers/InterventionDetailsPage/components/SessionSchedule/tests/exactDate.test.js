/**
 *
 * Tests for SessionSchedule
 *
 */

import React from 'react';
import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import 'jest-styled-components';
import configureStore from 'configureStore';
import { browserHistory } from 'react-router-dom';
import { Provider } from 'react-redux';

import { DEFAULT_LOCALE } from 'i18n';

import { SessionSchedule as SessionScheduleEnum } from 'models/Session';

import SessionSchedule from '../index';

describe('<SessionSchedule />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  const exactDateProps = {
    selectedScheduleOption: SessionScheduleEnum.EXACT_DATE,
    scheduleAt: new Date(2018, 11, 24, 10, 33, 30, 0),
  };

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <Provider store={store}>
          <SessionSchedule {...exactDateProps} />
        </Provider>
      </IntlProvider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const { container } = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <Provider store={store}>
          <SessionSchedule {...exactDateProps} />
        </Provider>
      </IntlProvider>,
    );
    expect(container).toMatchSnapshot();
  });
});
