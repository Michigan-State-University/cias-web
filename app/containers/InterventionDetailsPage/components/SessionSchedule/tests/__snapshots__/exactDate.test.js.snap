// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SessionSchedule /> Should render and match the snapshot 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: auto;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 8px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: auto;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: auto;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 24px;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: auto;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 10px;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: auto;
  width: 100%;
}

.c18 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: auto;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  justify-items: flex-end;
  margin-top: 10px;
  width: 100%;
}

.c3 {
  width: auto;
  height: auto;
  display: block;
  border-radius: 5px;
  position: relative;
  color: unset;
}

.c7 {
  width: auto;
  height: auto;
  display: block;
  border-radius: 5px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: unset;
}

.c8 {
  width: auto;
  height: auto;
  display: block;
  border-radius: 5px;
  color: unset;
}

.c6 {
  width: auto;
  height: auto;
  margin-left: 8px;
  cursor: pointer;
  -webkit-transition: -webkit-transform 0.2s;
  -webkit-transition: transform 0.2s;
  transition: transform 0.2s;
}

.c9 {
  width: auto;
  height: auto;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  padding-bottom: 24px;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.c1 {
  font-family: DM Sans;
  font-size: 13px;
  line-height: 130%;
  margin-bottom: 12px;
  font-size: 13px;
  color: rgba(47,56,80,0.6);
  color: rgba(47,56,80,0.6);
}

.c5 {
  font-family: DM Sans;
  font-size: 13px;
  line-height: 130%;
  font-weight: 700;
  color: rgba(183,39,234,1);
  cursor: pointer;
  color: rgba(183,39,234,1);
  cursor: pointer;
}

.c13 {
  font-family: DM Sans;
  font-size: 13px;
  line-height: 130%;
  margin-left: 5px;
  font-size: 15px;
  color: #2F3850;
}

.c19 {
  font-family: DM Sans;
  font-size: 13px;
  line-height: 130%;
  margin-left: 5px;
  font-size: 12px;
  color: rgba(210,55,29,1);
  color: rgba(210,55,29,1);
}

.c17 {
  padding: 12px;
  border-style: solid;
  border-width: 1px;
  border-color: #E9F1FB;
  border-radius: 5px;
  margin-right: 0;
  margin-left: 0;
  width: 130px;
  height: 50px;
  padding: 12px;
  font-size: 15px;
  color: rgba(47,56,80,1);
  text-align: left;
  color: rgba(47,56,80,1);
}

.c17:focus {
  box-shadow: none;
  outline: none;
  border-color: #107969;
}

.c17:disabled {
  color: #ABBBD1;
}

.c16 .schedule-date-picker {
  font-family: DM Sans;
  background-color: #E9F1FB;
}

.c16 .schedule-date-picker .react-datepicker__header {
  background-color: #F8FBFF;
}

.c16 .schedule-date-picker .react-datepicker__day {
  color: #2F3850;
}

.c16 .schedule-date-picker .react-datepicker__day:hover {
  color: #FFFFFF;
  background-color: rgba(200,102,234,0.7);
}

.c16 .schedule-date-picker .react-datepicker__day--keyboard-selected {
  background-color: #C866EA;
}

.c16 .schedule-date-picker .react-datepicker__day--in-range,
.c16 .schedule-date-picker .react-datepicker__day--in-selecting-range {
  color: #2F3850;
  background-color: rgba(200,102,234,0.1);
}

.c15 .schedule-date-picker .react-datepicker__day--selected:not(.c15 .schedule-date-picker .react-datepicker__day--in-selecting-range),
.c16 .schedule-date-picker .react-datepicker__day--range-start:not(.c16 .schedule-date-picker .react-datepicker__day--in-selecting-range),
.c16 .schedule-date-picker .react-datepicker__day--range-end:not(.c16 .schedule-date-picker .react-datepicker__day--in-selecting-range),
.c16 .schedule-date-picker .react-datepicker__day--selecting-range-start,
.c16 .schedule-date-picker .react-datepicker__day--selecting-range-end {
  color: #FFFFFF;
  background-color: #C866EA;
}

.c16 .schedule-date-picker .react-datepicker__year-option {
  background-color: #E9F1FB;
}

.c16 .schedule-date-picker .react-datepicker__month-option {
  background-color: #E9F1FB;
}

.c16 .schedule-date-picker .react-datepicker__time-list-item:hover {
  color: #FFFFFF;
  background-color: rgba(200,102,234,0.7) !important;
}

.c16 .schedule-date-picker .react-datepicker__time-list-item--selected {
  background-color: #C866EA !important;
}

.c16 .react-datepicker-wrapper {
  width: 100%;
}

@media only screen and (min-width:768px) {

}

@media only screen and (min-width:960px) {

}

@media only screen and (min-width:1170px) {

}

@media only screen and (min-width:1360px) {

}

@media only screen and (min-width:768px) {

}

@media only screen and (min-width:960px) {

}

@media only screen and (min-width:1170px) {

}

@media only screen and (min-width:1360px) {

}

<div>
  <div
    class="c0"
  >
    <p
      class="c1"
      color="#2F3850"
      font-size="13"
    >
      If this session is next in the sequence
    </p>
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <div
          class="c4"
          role="button"
        >
          <p
            class="c5"
            color="#B727EA"
            font-weight="bold"
          >
            Send email invitation on exact date
          </p>
          <img
            alt="arrow"
            class="c6"
            src="IMAGE_MOCK"
            transform=""
          />
        </div>
      </div>
      <div
        class="c7"
        display="flex"
      >
        <div
          class="c8"
          data-for="session-scheduling-cdh"
          data-tip=""
        >
          <img
            alt="?"
            class="c9"
            src="IMAGE_MOCK"
          />
        </div>
      </div>
    </div>
    <div
      class="c10"
    >
      <div
        class="c11"
      >
        <div
          class="c12"
        >
          <p
            class="c13"
            font-size="15"
          >
            Send 
            on exact date
          </p>
          <div
            class="c8"
          >
            <div
              class="c14"
              width="100%"
            >
              <div
                class="c11"
              >
                <div
                  class="c15 c16"
                >
                  <div
                    class="react-datepicker-wrapper"
                  >
                    <div
                      class="react-datepicker__input-container"
                    >
                      <input
                        aria-label="MM/DD/YYYY"
                        class="c17"
                        color="#2F3850"
                        dir="auto"
                        font-size="15"
                        height="50"
                        placeholder="MM/DD/YYYY"
                        type="text"
                        value="12/24/2018"
                        inputmode="numeric"
                        width="130"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="c18"
          width="100%"
        >
          <p
            class="c19"
            color="#D2371D"
            font-size="12"
          >
            *The information will be sent at 10:00 am
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
